src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -67,11 +67,23 @@ internal class SourceLocalSymbol : LocalSymbol
             _locations = ImmutableArray.Create<Location>(identifierToken.GetLocation());
         }
 
-        internal Binder Binder
+        /// <summary>
+        /// Binder that owns the scope for the local, the one that returns it in its <see cref="Binder.Locals"/> array.
+        /// </summary>
+        internal Binder ScopeBinder
         {
-            get { return binder; }
+            get { return _scopeBinder; }
         }
 
+        /// <summary>
+        /// Binder that should be used to bind type syntax for the local.
+        /// </summary>
+        internal Binder TypeSyntaxBinder
```

Comment:
It wasn't absolutely necessary, but felt appropriate to me, given the nature of the other changes I was making. Plus I didn't think you had any changes in flight. Sorry about the inconvenience this might have caused you. I hope the conflicts wouldn't be hard to deal with.